### DISCUSSION TITLE
Build Error When Trying To Deploy - ModuleNotFoundError: No module named '_sqlite3'

### DISCUSSION THREAD

**Discussion by ah410** [vercel/community#4971] [Help] [answered]
### Summary

I made a python flask application and am trying to deploy it to Vercel but am encountering a building error when I click deploy. I specifically added gunicorn to the install command because it seemed to not install gunicorn in my requirements.txt even though I included it. 

Build Command: `gunicorn -w 4 -b 0.0.0.0:8000 app:app`

Install Command: `pip install -r requirements.txt gunicorn`

The build starts up with gunicorn but I encounter a `ModuleNotFoundError`. 
`ModuleNotFoundError: No module named '_sqlite3'`

 I've ran it locally on my system and it works. I'm using python version 3.10.12 locally on my Ubuntu system. I added the docs below for the build logs. First traceback is on page 4. I tried using Pipfile according to the vercel documentation, but no luck. https://vercel.com/docs/functions/serverless-functions/runtimes/python

[logs.docx](https://github.com/vercel/community/files/13465817/logs.docx)

### Example

![image](https://github.com/vercel/community/assets/115514687/4b935f9a-e588-4bca-98ca-f838c3bd2d73)

### Steps to Reproduce

You can reproduce the issue by cloning my CineVerse repository and trying to deploy to vercel using the commands below. Then, hit deploy in vercel and it will generate build logs.

Build Command: `gunicorn -w 4 -b 0.0.0.0:8000 app:app`

Install Command: `pip install -r requirements.txt gunicorn`

**Comment by miguel-ossa** [comment, marked as answer]
You don't need to modify any install or build commands. Put your code inside an "api" folder, like I did in this project recently:

[https://github.com/miguel-ossa/tresMercadosBlog](https://github.com/miguel-ossa/tresMercadosBlog)

[https://tres-mercados-blog.vercel.app/](https://tres-mercados-blog.vercel.app/)

Also, create a simple "vercel.json" file, pointint at your main module (in my case is "main").

I hope it helps. About the sqlite3 error... you'll have to use another kind of db, like f.e. Postgres.

![image](https://github.com/vercel/community/assets/9202841/4b8b965c-34f8-4d50-a4bc-65b91bcacfda)

And create some environment vars:

![image](https://github.com/vercel/community/assets/9202841/676b1ce9-0de9-4af9-af00-5498945b9554)

Ooops... you also don't need gunicorn, like in other hostings.

**Reply by ah410** [reply]
Ok, thanks! I'll give that a try, switch to Postgres, and then redeploy and see if it works. It might take a few days though to migrate to Postgres since I'm busy with finals coming up.

**Reply by ah410** [reply]
I marked the question down as answered. It works locally on my system right now, I just have to iron out some postgres sql statements that I converted from sqlite statements that aren't working as intended. 

The postgreSQL db works, put my code in api folder, and added vercel.json to specifiy the entry point(for me it's api/app.py). This is local right now till I fix my postgres SQL statements. 

So, once I get that finished, I just keep default build and install commands and it should work? Is there really no need to specify the install command of `pip install -r requirements.txt`?

Edit: I had more free time than I thought today so I just grinded out as much as I could.

**Reply by miguel-ossa** [reply]
The platform does the build automatically from your requirements.txt, so you have to do nothing special.

![image](https://github.com/vercel/community/assets/9202841/267c1966-6460-451b-ace5-53e4d79d3d4e)

**Reply by ah410** [reply]
So, I finally finished with my postgres statements and it works as intended. I still get no website when I deploy though. It's a 404 error.

Edit: I'm tearing my hair out at trying to deploy my website. I've been trying for months now with zero luck. It feels like black magic trying to get a simple application to deploy so recruiters can see my website. I don't understand what I did wrong. I followed all the steps. I created a vercel.json, used default build and install commands, put my code in /api, and switched to vercel's postgres database.
